pkg/cmd/codespace/ssh.go

```diff
@@ -236,48 +234,74 @@ func (a *App) SSH(ctx context.Context, sshArgs []string, opts sshOptions) (err e
 	}
 }
 
-func useAutomaticSSHKeys(
+// selectSSHKeys evaluates available key pairs and select which should be used to connect to the codespace
+// using the precedence rules below. If there is no error, a keypair is always returned and additionally a
+// bool flag is returned to specify if the private key need be appended to the ssh arguments (it doesn't need
+// to be if the key was selected from a -i argument).
+//
+// Precedence rules:
+// 1. Key which is specified by -i
+// 2. Automatic key, if it already exists
+// 3. First valid keypair in ssh config (according to ssh -G)
+// 4. Automatic key, newly created
+func selectSSHKeys(
 	ctx context.Context,
 	sshContext ssh.Context,
-	apiClient apiClient,
 	args []string,
 	opts sshOptions,
-) (bool, error) {
+) (*ssh.KeyPair, bool, error) {
 	customConfigPath := ""
 	for i := 0; i < len(args); i += 1 {
 		arg := args[i]
 
 		if arg == "-i" {
-			if i+1 < len(args) && path.Base(args[i+1]) == automaticPrivateKeyName {
-				return true, nil
+			if i+1 >= len(args) {
+				return nil, false, errors.New("missing value to -i argument")
 			}
 
-			// User specified a custom identity file so just trust it is correct
-			return false, nil
+			// User manually specified an identity file so just trust it is correct
+			return &ssh.KeyPair{
+				PrivateKeyPath: args[i+1],
+				PublicKeyPath:  args[i+1] + ".pub",
+			}, false, nil
 		}
 
-		if arg == "-F" && i < len(args)-1 {
+		if arg == "-F" && i+1 < len(args) {
 			// ssh only pays attention to that last specified -F value, so it's correct to overwrite here
 			customConfigPath = args[i+1]
 		}
 	}
 
-	if automaticKeySSHKeysExist(sshContext) {
+	if autoKeyPair := automaticSSHKeyPair(sshContext); autoKeyPair != nil {
 		// If the automatic keys already exist, just use them
-		return true, nil
+		return autoKeyPair, true, nil
+	}
+
+	configKeyPair, err := firstConfiguredKeyPair(ctx, customConfigPath, opts.profile)
+	if err != nil && !errors.Is(err, errKeyFileNotFound) {
+		return nil, false, fmt.Errorf("checking configured keys: %w", err)
+	}
+
+	// Even if there was no error, there may simply be no configured keys, so still check for nil here
```

Comment:
The comment is no longer true, right?

Also this section of code is not passing the squinting test. Here's how I'd structure this block of code:
```go
keyPair, err := firstConfiguredKeyPair(ctx, customConfigPath, opts.profile)
if err != nil {
	if !errors.Is(err, errKeyFileNotFound) {
		return nil, false, fmt.Errorf("checking configured keys: %w", err)
	}
	
	// key not found, generate
	keyPair, err = generateAutomaticSSHKeys(sshContext)
	if err != nil {
		return nil, false, fmt.Errorf("generating automatic keypair: %w", err)
	}
}
return keyPair, true, nil
```

Reply:
>The comment is no longer true, right?

Oops 😄 removed

> Also this section of code is not passing the squinting test

Updated accordingly